js/errors.js

```diff
@@ -282,3 +282,32 @@
 
     NoConnectionError.prototype = Object.create(Error.prototype);
     NoConnectionError.prototype.constructor = NoConnectionError;
+
+
+    /**
+     * Log the error object to the given ermrest location.
+     * It will generate a put request to the /terminal_error with the correct headers.
+     * ermrset will return a 400 page, but will log the message.
+     * @param  {object} err             the error object
+     * @param  {string} ermrestLocation the ermrest location
+     */
+    module.logError = function (err, ermrestLocation) {
+        var defer = module._q.defer();
+        var http = module._wrap_http(module._http);
+
+        var headers = {};
+        headers[module._contextHeaderName] = {
+            e: 1,
```

Comment:
What does `e: 1` mean? Is that the error code attached to the terminal error?

Will that always be the error code or do we want it to be the one returned for that request?